thunder/dynamo/compiler.py

```diff
@@ -18,13 +18,29 @@
 from thunder.dynamo.splitter import _splitter
 from thunder.core.utils import check
 from thunder.dynamo.benchmark_utils import ThunderCompileSpecification
+from thunder.transforms.extraction_only_prologue_transform import ExtractionOnlyPrologueTransform
 
 if TYPE_CHECKING:
     from thunder.dynamo.utils import SubgraphInfo
+    from thunder.core.transform_common import Transform
     from os import PathLike
     from collections.abc import Callable
 
 
+def _add_prologue_pruning(options: dict):
+    """
+    Add a transform to prune prologue checks to the list of transforms in the given options dictionary.
+
+    Args:
+        options: The dictionary of options to modify
+    """
+    transforms: list[Transform] | None = options.get("transforms", None)
+    if transforms is None:
+        transforms = []
+    transforms.append(ExtractionOnlyPrologueTransform())
```

Comment:
`thunder.jit` allows passing transforms sequence as a tuple. If a tuple is passed here the append method won't work.

Reply:
As per the docs, transforms should be a list of Transforms or None (though I don't think this is verified at runtime).

https://github.com/Lightning-AI/lightning-thunder/blob/7c16a1ad198e59f823f6174c059c693c54589653/thunder/__init__.py#L290

https://github.com/Lightning-AI/lightning-thunder/blob/7c16a1ad198e59f823f6174c059c693c54589653/thunder/__init__.py#L317